frontend/helpers/next-helpers.ts

```diff
@@ -24,16 +24,42 @@ export const withLogging: GetServerSidePropsMiddleware = (next) => {
    });
 };
 
-export function noindexDevDomains(context: GetServerSidePropsContext) {
-   if (context.req.headers['user-agent'] !== PROD_USER_AGENT) {
-      context.res.setHeader('X-Robots-Tag', 'noindex, nofollow');
-   }
+type ContextType = GetServerSidePropsContext;
+
+export type GetRestrictRobots = (
+   context: GetServerSidePropsContext
+) => RestrictRobots | undefined;
+
+enum RestrictRobots {
+   RESTRICT_ALL = 'noindex, nofollow, nosnippet, noarchive, noimageindex',
+   RESTRICT_INDEXING = 'noindex, follow, nosnippet, noarchive, noimageindex',
+   RESTRICT_FOLLOWING = 'index, nofollow',
+   ALLOW_ALL = 'index, follow',
 }
 
-export const withNoindexDevDomains: GetServerSidePropsMiddleware = (next) => {
-   return (context) => {
-      const result = next(context);
-      noindexDevDomains(context);
-      return result;
+export function withRobotsHeader(
+   getRestrictRobots: GetRestrictRobots
+): GetServerSidePropsMiddleware {
+   return (next) => (context) => {
+      maybeSetRobotsHeader(context, getRestrictRobots);
+      return next(context);
    };
-};
+}
+
+function maybeSetRobotsHeader(
+   context: ContextType,
+   maybeGetRestrictRobots: GetRestrictRobots
+) {
+   const restrictRobots = maybeGetRestrictRobots(context);
+   if (restrictRobots) {
+      context.res?.setHeader('X-Robots-Tag', restrictRobots);
+   }
+}
+
+export function noindexDevDomains(context: ContextType) {
+   if (context.req.headers['user-agent'] !== PROD_USER_AGENT) {
+      // return RestrictRobots.RESTRICT_ALL;
+   }
+
+   return undefined;
```

Comment:
```suggestion
   return RestricRobot.ALLOW_ALL;
```

Reply:
Doesn't that apply to this change regardless of which way this goes (i.e. it needs to line up?)

Reply:
Fair point. That's why I started refactoring it. The hope was we could have a more general way of getting the indexing string out of a shared function so both the headers and meta tags can reused the code.